facilitator/src/transport/gcs.rs

```diff
@@ -373,98 +383,81 @@ impl StreamingTransferWriter {
             .timeout_read(10_000) // ten seconds
-            .send(body);
+            .send_bytes(&self.buffer);
 
         // On success we expect HTTP 308 Resume Incomplete and a Range: header,
         // unless this is the last part and the server accepts the entire
         // provided Content-Range, in which case it's HTTP 200, or 201 (?).
         // https://cloud.google.com/storage/docs/performing-resumable-uploads#chunked-upload
         match http_response.status() {
-            200 | 201 => {
-                if !last_chunk {
-                    return Err(anyhow!(
-                        "received HTTP 200 or 201 response with chunks remaining"
-                    ));
-                }
+            200 | 201 if last_chunk => {
+                // Truncate the buffer to "drain" it of uploaded bytes
+                self.buffer.truncate(0);
+                Ok(())
             }
+            200 | 201 => Err(anyhow!(
+                "received HTTP 200 or 201 response with chunks remaining"
+            )),
+            308 if !http_response.has("Range") => Err(anyhow!(
+                "No range header in response from GCS: {:?}",
+                http_response.into_string()
+            )),
             308 => {
-                if !http_response.has("Range") {
-                    return Err(anyhow!(
-                        "No range header in response from GCS: {:?}",
-                        http_response.into_string()
-                    ));
-                }
                 let range_header = http_response.header("Range").unwrap();
-                // The range header is like "bytes=111-222", and represents a
-                // range in the overall object, not the current chunk.
-                let range = range_header
-                    .strip_prefix("bytes=")
-                    .context(format!("unexpected Range header {}", range_header))?;
-                let mut values = range.splitn(2, '-');
-                let start = values
-                    .next()
-                    .context(format!("unexpected Range header {}", range_header))?
+                // The range header is like "bytes=0-222", and represents the
+                // uploaded portion of the overall object, not the current chunk
+                let end = range_header
+                    .strip_prefix("bytes=0-")
+                    .context(format!(
+                        "Range header {} missing bytes prefix",
+                        range_header
+                    ))?
                     .parse::<usize>()
-                    .context(format!("unexpected Range header {}", range_header))?;
-                let end = values
-                    .next()
-                    .context(format!("unexpected Range header {}", range_header))?
-                    .parse::<usize>()
-                    .context(format!("unexpected Range header {}", range_header))?;
-
-                if start != self.object_upload_position {
```

Comment:
It looks like the expected semantics of the `start` part of the range changed in the latest commit, from expecting it to be object_upload_position to always being 0. It seems like one version or the other should have resulted in throwing errors?